File: code/DataTypeExample.java
public class DataTypeExample {
    public static void main(String[] args) {
        // myNum = 5; // this doesn't work
        int myNum;  // Declare a integer varaible
        myNum = 5; // Assign Integer (whole number)
        
        // myNum = 1.234; // This does not work

        float myFloatNum = 5.99f; // Floating point number
        double myDoubleNum = 5.99999999999; // Double
        char myLetter = 'D'; // Character
        boolean myBool = true; // Boolean
        String myText = "Hello"; // String

        System.out.println("My number: " + myNum + ", " + "My float: " + myFloatNum + ", " + "My double: " + myDoubleNum
                + ", " + "My letter: " + myLetter + ", " + "My Bool: " + myBool + ", " + "My Text: " + myText);

        // myNum = 5.123; // this doesn't work
        // myBool = "True"; // this doesn't work

        myDoubleNum = myNum; // this works
        
        System.out.println("My number: " + myNum + ", " + "My float: " + myFloatNum + ", " + "My double: " + myDoubleNum
                + ", " + "My letter: " + myLetter + ", " + "My Bool: " + myBool + ", " + "My Text: " + myText);

    }
}

File: code/data_type_example.py

def main():
    myNum = int(5) 
    myFloatNum = 5.99
    myDoubleNum = 5.99999999999
    myLetter = 'D'
    myBool = True
    myText = "Hello"

    print("My number: ", myNum, ", ", "My float: ", myFloatNum, ", ", "My double: ",
     myDoubleNum, ", ", "My letter: " , myLetter , ", " , "My Bool: " , myBool , ", " , "My Text: " , myText)

    print(f"Before new assignment, the type of myNum value is {myNum} and type is {type(myNum)}")
    myNum = 5.123   # this works
    print(f"After new assignment, the type of myNum value is {myNum} and type is {type(myNum)}")
    print(f"Before new assignment, the type of myNum value is {myBool} and type is {type(myBool)}")
    myBool = "True" # this works 
    print(f"Before new assignment, the type of myNum value is {myBool} and type is {type(myBool)}")

    myDoubleNum = myNum  # this is ok 

if __name__ == "__main__":
    main()